Tests/FluentAssertions.Specs/AssertionOptionsSpecs.cs

```diff
@@ -14,387 +14,380 @@ namespace FluentAssertions.Specs;
 
 public class AssertionOptionsSpecs
 {
-    [Fact]
-    public void When_injecting_a_null_configurer_it_should_throw()
+    public abstract class Given_temporary_global_assertion_options : GivenWhenThen
     {
-        // Arrange
-        Action act = () => AssertionOptions.AssertEquivalencyUsing(defaultsConfigurer: null);
+        protected override void Dispose(bool disposing)
+        {
+            AssertionOptions.AssertEquivalencyUsing(_ => new EquivalencyAssertionOptions());
 
-        // Assert
-        act.Should().ThrowExactly<ArgumentNullException>()
-            .WithParameterName("defaultsConfigurer");
+            base.Dispose(disposing);
+        }
     }
-}
 
-// Due to tests that call AssertionOptions
-[CollectionDefinition("AssertionOptionsSpecs", DisableParallelization = true)]
```

Comment:
After checking this branch out, I got this Exception, so I think having `DisableParallelization = true` is still needed.
```
Expected a <Xunit.Sdk.XunitException> to be thrown, but found <System.InvalidOperationException>: "
"System.InvalidOperationException with message "Collection was modified; enumeration operation may not execute."
     at System.Collections.Generic.List`1.Enumerator.MoveNextRare()
     at FluentAssertions.Equivalency.EquivalencyValidator.RunStepsUntilEquivalencyIsProven(Comparands comparands, IEquivalencyValidationContext context) in C:\dev\fluentassertions\Src\FluentAssertions\Equivalency\EquivalencyValidator.cs:line 69
     at FluentAssertions.Equivalency.EquivalencyValidator.RecursivelyAssertEquality(Comparands comparands, IEquivalencyValidationContext context) in C:\dev\fluentassertions\Src\FluentAssertions\Equivalency\EquivalencyValidator.cs:line 40
     at FluentAssertions.Equivalency.Steps.EnumerableEquivalencyValidator.TryToMatch[T](Object subject, T expectation, Int32 expectationIndex) in C:\dev\fluentassertions\Src\FluentAssertions\Equivalency\Steps\EnumerableEquivalencyValidator.cs:line 181
     at FluentAssertions.Equivalency.Steps.EnumerableEquivalencyValidator.LooselyMatchAgainst[T](IList`1 subjects, T expectation, Int32 expectationIndex) in C:\dev\fluentassertions\Src\FluentAssertions\Equivalency\Steps\EnumerableEquivalencyValidator.cs:line 150
     at FluentAssertions.Equivalency.Steps.EnumerableEquivalencyValidator.AssertElementGraphEquivalencyWithLooseOrdering[T](Object[] subjects, T[] expectations) in C:\dev\fluentassertions\Src\FluentAssertions\Equivalency\Steps\EnumerableEquivalencyValidator.cs:line 121
     at FluentAssertions.Equivalency.Steps.EnumerableEquivalencyValidator.AssertElementGraphEquivalency[T](Object[] subjects, T[] expectations, INode currentNode) in C:\dev\fluentassertions\Src\FluentAssertions\Equivalency\Steps\EnumerableEquivalencyValidator.cs:line 85
     at FluentAssertions.Equivalency.Steps.EnumerableEquivalencyValidator.Execute[T](Object[] subject, T[] expectation) in C:\dev\fluentassertions\Src\FluentAssertions\Equivalency\Steps\EnumerableEquivalencyValidator.cs:line 43
     at FluentAssertions.Equivalency.Steps.GenericEnumerableEquivalencyStep.HandleImpl[T](EnumerableEquivalencyValidator validator, Object[] subject, IEnumerable`1 expectation) in C:\dev\fluentassertions\Src\FluentAssertions\Equivalency\Steps\GenericEnumerableEquivalencyStep.cs:line 60
  --- End of stack trace from previous location ---
     at FluentAssertions.Equivalency.Steps.GenericEnumerableEquivalencyStep.Handle(Comparands comparands, IEquivalencyValidationContext context, IEquivalencyValidator nestedValidator) in C:\dev\fluentassertions\Src\FluentAssertions\Equivalency\Steps\GenericEnumerableEquivalencyStep.cs:line 52
     at FluentAssertions.Equivalency.EquivalencyValidator.RunStepsUntilEquivalencyIsProven(Comparands comparands, IEquivalencyValidationContext context) in C:\dev\fluentassertions\Src\FluentAssertions\Equivalency\EquivalencyValidator.cs:line 71
     at FluentAssertions.Equivalency.EquivalencyValidator.RecursivelyAssertEquality(Comparands comparands, IEquivalencyValidationContext context) in C:\dev\fluentassertions\Src\FluentAssertions\Equivalency\EquivalencyValidator.cs:line 40
     at FluentAssertions.Equivalency.EquivalencyValidator.AssertEquality(Comparands comparands, EquivalencyValidationContext context) in C:\dev\fluentassertions\Src\FluentAssertions\Equivalency\EquivalencyValidator.cs:line 22
     at FluentAssertions.Collections.GenericCollectionAssertions`3.BeEquivalentTo[TExpectation](IEnumerable`1 expectation, Func`2 config, String because, Object[] becauseArgs) in C:\dev\fluentassertions\Src\FluentAssertions\Collections\GenericCollectionAssertions.cs:line 369
     at FluentAssertions.Collections.GenericCollectionAssertions`3.NotBeEquivalentTo[TExpectation](IEnumerable`1 unexpected, Func`2 config, String because, Object[] becauseArgs) in C:\dev\fluentassertions\Src\FluentAssertions\Collections\GenericCollectionAssertions.cs:line 1751
     at FluentAssertions.Collections.GenericCollectionAssertions`3.NotBeEquivalentTo[TExpectation](IEnumerable`1 unexpected, String because, Object[] becauseArgs) in C:\dev\fluentassertions\Src\FluentAssertions\Collections\GenericCollectionAssertions.cs:line 1713
     at FluentAssertions.Specs.Collections.CollectionAssertionSpecs.NotBeEquivalentTo.<>c__DisplayClass1_0.<When_large_collection_is_equivalent_to_another_equally_size_collection_it_should_throw>b__0() in C:\dev\fluentassertions\Tests\FluentAssertions.Specs\Collections\CollectionAssertionSpecs.BeEquivalentTo.cs:line 183
     at FluentAssertions.Specialized.ActionAssertions.InvokeSubject() in C:\dev\fluentassertions\Src\FluentAssertions\Specialized\ActionAssertions.cs:line 25
     at FluentAssertions.Specialized.DelegateAssertions`2.InvokeSubjectWithInterception() in C:\dev\fluentassertions\Src\FluentAssertions\Specialized\DelegateAssertions.cs:line 221
```

Reply:
Hmm, indeed. It does fail intermittently. But I don't actually understand why this is needed. All the tests in `AssertionOptionsSpecs` have now been marked with `[Collection("AssertionOptionsSpecs")]`, so none of them should run in parallel. Why